contracts/ValueGuardsV2.sol

```diff
@@ -0,0 +1,173 @@
+// SPDX-License-Identifier: Apache 2.0
+pragma solidity =0.8.25;
+
+import { OwnableUpgradeable } from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+
+abstract contract ValueGuardsV2 is OwnableUpgradeable {
+    // =============================================== Storage ========================================================
+
+    /**
+     * @dev Represents a parameter range for a specific getter function.
+     * @param getter The getter function signature.
+     * @param range The range of values for the parameter.
+     */
+    struct ParameterRange {
+        bytes4 getter;
+        uint256[] range;
+    }
+
+    struct ValueGuardsStorage {
+        /**
+        * @dev A mapping that stores the allowed parameter ranges for each function signature.
+        */
+        mapping(bytes4 => ParameterRange) allowedParameterRange;
+    }
+
+    bytes32 private constant VALUEGUARDS_STORAGE_NAMESPACE = keccak256("valueguards.storage");
```

Comment:
We should use ERC-7201 defined storage location formula here: `keccak256(abi.encode(uint256(keccak256(bytes(id))) - 1)) & ~bytes32(uint256(0xff))`.

Reply:
Using in [9a32adc](https://github.com/DMDcoin/diamond-contracts-core/pull/264/commits/9a32adcd012ef2dd54fac637d0afdeef3ae80163)